credmon/utils/utils.py

```diff
@@ -256,26 +257,40 @@ def generate_secret_key():
 
     keyfile = os.path.join(htcondor.param.get("SEC_CREDENTIAL_DIRECTORY", "/var/lib/condor/credentials"), "wsgi_session_key")
 
+    # Create the secret key file, if possible, with read-only permissions, if it doesn't exist
     try:
-        fd = os.open(keyfile, os.O_CREAT | os.O_RDWR, 0o600)
-        current_key = os.read(fd, 24)
-    except Exception:
-        logger.warning("Unable to access WSGI session key; will use a non-persistent key")
+        os.close(os.open(keyfile, os.O_CREAT | os.O_EXCL | os.O_RDWR, stat.S_IRUSR))
+    except OSError as os_error:
+        # An exception will be thrown if the file already exists, and that's fine and good.
+        if not (os_error.errno == errno.EEXIST):
+            logger.warning("Unable to access WSGI session key at %s (%s);  will use a non-persistent key.", keyfile, str(os_error))
+            return os.urandom(16)
+
+    # Open the secret key file.
+    try:
+        with open(keyfile, 'rb') as f:
+            current_key = f.read(24)
+    except Exception as e:
```

Comment:
Utilize the more specific `IOError`.

Reply:
Resolved in https://github.com/htcondor/scitokens-credmon/pull/21/commits/298b4e3ad35168895aaa99ea8855b1722fb703df